localstack/utils/aws/request_context.py

```diff
@@ -63,6 +65,12 @@ def extract_region_from_auth_header(headers):
     return region
 
 
+def extract_region_from_host_header(headers):
+    host = headers.get("Host") or ""
+    region = re.search(AWS_REGION_REGEX, host, flags=re.IGNORECASE)
+    return region.group() if region else None
```

Comment:
I think this might not be necessary if the routing rules are adjusted a bit (see other comment).

Reply:
Yes, you're right. after editing the routes it became unnecessary.